integration_test/tests/free_and_cancel_tests.rs

```diff
@@ -56,6 +56,82 @@ mod integration {
         }
     }
 
+    unsafe fn exec_direct_default_query(stmt_handle: HStmt) {
+        let mut query: Vec<WideChar> = cstr::to_widechar_vec("SELECT * FROM integration_test.foo");
+        query.push(0);
+        assert_eq!(
+            SqlReturn::SUCCESS,
+            SQLExecDirectW(stmt_handle, query.as_ptr(), SQL_NTS as i32),
+            "{}",
+            get_sql_diagnostics(HandleType::SQL_HANDLE_STMT, stmt_handle as Handle)
+        );
+    }
+
+    /// This test is inspired by the SSIS Query flow. It is altered to be
+    /// more general than that specific flow, with a focus on freeing the
+    /// statement handle after calling SQLBindCol. This flow depends on
+    /// default_setup_connect_and_alloc_stmt.
+    /// After allocating a statement handle, the flow is:
+    ///     - SQLSetStmtAttrW(SQL_ATTR_ROW_BIND_TYPE, SQL_BIND_BY_COLUMN)
+    ///     - SQLSetStmtAttrW(SQL_ATTR_ROW_ARRAY_SIZE, 1000)
+    ///     - SQLExecDirectW(<query>)
+    ///     - <loop: until SQLFetchScroll returns SQL_NO_DATA>
+    ///         - SQLBindCol for each column
+    ///         - SQLFetchScroll
+    ///     - SQLFreeStmt(SQL_UNBIND)
+    ///     - SQLDisconnect
+    ///     - SQLFreeHandle(SQL_HANDLE_DBC)
+    ///     - SQLFreeHandle(SQL_HANDLE_ENV)
+    #[test]
+    fn test_free_stmt_after_bind_col() {
```

Comment:
In the spirit of having test files associated with the functionality we are testing, I think that the logic around retrieving data from bound columns should move to a different file.
The focus of this function, if I understand correctly, is to check that we unbind bound columns correctly.
I would remove the portion of the code setting the SQL_BIND_BY_COLUMN (this is the default value) and SQL_ATTR_ROW_ARRAY_SIZE attributes, it doesn't matter here and will likely not work as expected when binding rowset is supported because it is asking for 1000 rows and the buffer size is 4, so I'd expect SQL_FETCH to return SQL_ERROR or SQL_SUCCESS_WITH_INFO, but likely not SQL_SUCCESS or SQL_NO_DATA.

We should also verify that we bind buffers and put the data in them correctly, but this shouldn't happen here.

For checking free behavior, this is enough :
   ///    - SQLExecDirectW(<query>)
   ///    - SQLBindCol
    ///   - SQLFetchScroll
    ///    - SQLFreeStmt(SQL_UNBIND)
    
    
  For checking rowset retrieval, you'll need :
   ///    - SQLSetStmtAttrW(SQL_ATTR_ROW_ARRAY_SIZE, xxx)
   ///    - SQLExecDirectW(<query>)
   ///    - SQLBindCol
   ///   - <loop: until SQLFetchScroll returns SQL_NO_DATA> <--- Want to verify that SQLFetchScroll returns data for the expected number of times
   ///             - SQLFetchScroll.  <--- Want to check that the buffer content changed (a.k.a new data was copied)